engine/execution/storehouse/checkpoint_validator.go

```diff
@@ -0,0 +1,137 @@
+package storehouse
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rs/zerolog"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/onflow/flow-go/engine/execution"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/ledger/complete/wal"
+	"github.com/onflow/flow-go/storage"
+)
+
+// ValidateWithCheckpoint validates the registers in the given store against the leaf nodes read from the checkpoint file.
+// Limitation: the validation can not cover if there are extra non-empty registers in the store that are not in the checkpoint file.
```

Comment:
Is it possible to get all payloads of the given height from the storehouse?  If so, we can check if storehouse has any payloads that don't exist in the trie.

Reply:
Yeah, we can implement it separately. To get all payloads of the given height, we would have to iterate through the entire key space, and check if any key does not exist in the trie _and_ also was updated at the given height or lower. Will do in future PR